File: C-Sharp/GetItems.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
using VB = Microsoft.VisualBasic;
namespace _4PosBackOffice.NET
{
	static class GetIndex
	{
		public static object GetIndexer(ref object text1, ref IEnumerable<Control> text2)
		{
			int index = 0;
			Control m = new Control();
			index = 0;
			foreach (Control m_loopVariable in text2) {
				m = m_loopVariable;
				if (m.Name != text1.Name) {
					index = index + 1;
				} else {
					break; // TODO: might not be correct. Was : Exit For
				}
			}
			return index;
		}
	}
}

File: VB/4PosBackOffice.NET/GetItems.vb
Option Strict Off
Option Explicit On
Imports VB = Microsoft.VisualBasic
Module GetIndex
    Public Function GetIndexer(ByRef text1 As Object, ByRef text2 As IEnumerable(Of Control))
        Dim index As Integer
        Dim m As New Control
        index = 0
        For Each m In text2
            If m.Name <> text1.Name Then
                index = index + 1
            Else
                Exit For
            End If
        Next
        Return index
    End Function
End Module
